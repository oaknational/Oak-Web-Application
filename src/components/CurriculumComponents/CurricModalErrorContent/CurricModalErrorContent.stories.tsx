import { Meta, StoryObj } from "@storybook/react";
import { OakThemeProvider, oakDefaultTheme } from "@oaknational/oak-components";

import Component from "./CurricModalErrorContent";

const meta: Meta<typeof Component> = {
  component: Component,
  decorators: [
    (Story) => (
      <OakThemeProvider theme={oakDefaultTheme}>
        <Story />
      </OakThemeProvider>
    ),
  ],
};

export default meta;
type Story = StoryObj<typeof Component>;

export const CurricModalErrorContent: Story = {
  args: {
    statusCode: "404",
    message: "Something went wrong",
    additional: "Close the modal to continue",
  },
  render: (args) => <Component {...args} />,
};
