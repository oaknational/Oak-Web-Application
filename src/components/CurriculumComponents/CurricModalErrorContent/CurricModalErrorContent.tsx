import { OakFlex, OakHeading, OakP } from "@oaknational/oak-components";
import styled from "styled-components";

const shadow =
  "-1px -1px 0 #000, 1px -1px 0 #000, -1px 1px 0 #000, 1px 1px 0 #000";

const ErrorHeading = styled(OakHeading)`
  color: white;
  font-size: 120px;
  line-height: 144px;
  text-shadow: ${shadow};
`;

type CurricModalErrorContentProps = {
  statusCode: string;
  message: string;
  additional?: string;
};
export default function CurricModalErrorContent({
  statusCode,
  message,
  additional,
}: CurricModalErrorContentProps) {
  return (
    <OakFlex
      $gap="space-between-s"
      $mv="space-between-l"
      $flexDirection={"column"}
    >
      <ErrorHeading tag="h1">{statusCode}</ErrorHeading>
      <OakHeading $font={"heading-4"} tag="h2">
        {message}
      </OakHeading>
      {additional && <OakP $font={"body-1"}>{additional}</OakP>}
    </OakFlex>
  );
}
