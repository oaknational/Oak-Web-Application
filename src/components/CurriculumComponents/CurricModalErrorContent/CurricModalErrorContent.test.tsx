import CurricModalErrorContent from "./CurricModalErrorContent";

import renderWithTheme from "@/__tests__/__helpers__/renderWithTheme";

describe("CurricModalErrorContent", () => {
  it("should render correctly", () => {
    const { container } = renderWithTheme(
      <CurricModalErrorContent statusCode="404" message="Missing!" />,
    );
    expect(container).toMatchSnapshot();
  });

  it("should render additional when supplied", () => {
    const { container } = renderWithTheme(
      <CurricModalErrorContent
        statusCode="404"
        message="Missing!"
        additional="Close this modal"
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
