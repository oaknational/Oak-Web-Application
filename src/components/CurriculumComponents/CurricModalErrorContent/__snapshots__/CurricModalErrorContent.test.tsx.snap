// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurricModalErrorContent should render additional when supplied 1`] = `
<div>
  <div
    class="sc-fqkvVR sc-kAyceB kRjBvS FywIr"
  >
    <h1
      class="sc-kpDqfm sc-c9371ddb-0 jeWvUz hmfNXu"
    >
      404
    </h1>
    <h2
      class="sc-kpDqfm cCIhAo"
    >
      Missing!
    </h2>
    <p
      class="sc-dLMFU eFqtPi"
    >
      Close this modal
    </p>
  </div>
</div>
`;

exports[`CurricModalErrorContent should render correctly 1`] = `
<div>
  <div
    class="sc-fqkvVR sc-kAyceB kRjBvS FywIr"
  >
    <h1
      class="sc-kpDqfm sc-c9371ddb-0 jeWvUz hmfNXu"
    >
      404
    </h1>
    <h2
      class="sc-kpDqfm cCIhAo"
    >
      Missing!
    </h2>
  </div>
</div>
`;
